src/llm_models/openai.rs

```diff
@@ -121,9 +132,11 @@ impl LLMModel for OpenAIModels {
     fn function_call_default(&self) -> bool {
         //OpenAI documentation: https://platform.openai.com/docs/guides/gpt/function-calling
         match self {
-            OpenAIModels::TextDavinci003 | OpenAIModels::Gpt3_5Turbo | OpenAIModels::Gpt4_32k => {
-                false
-            }
+            OpenAIModels::TextDavinci003
+            | OpenAIModels::Gpt3_5Turbo
+            | OpenAIModels::Gpt4_32k
+            | OpenAIModels::O1Preview
+            | OpenAIModels::O1Mini => false,
```

Comment:
o1 models do not support function calling (or any tools) for now.

Reply:
got it